src/main/java/org/springframework/integration/aws/support/S3Session.java

```diff
@@ -82,8 +82,11 @@ public S3Object[] list(String path) {
 		List<S3Object> objectSummaries = new ArrayList<>();
 		do {
 			objectListing = this.amazonS3.listObjects(listObjectsRequest.build());
-			objectSummaries.addAll(objectListing.contents());
-			listObjectsRequest.marker(objectListing.nextMarker());
+			List<S3Object> contents = objectListing.contents();
+			objectSummaries.addAll(contents);
+			if (objectListing.isTruncated()) {
```

Comment:
Do we really need this `if` here?
Why we just cannot always set that last key as `marker` and rely on the next `while (objectListing.isTruncated())`?

Reply:
this condition is to ensure that list is not empty, it can be change to `!objectListing.isEmpty()`. Any thoughts?

Reply:
I see. OK.
Please, add your name to the `@author` list of this class and I'll merge it shortly.

Thanks